projects/ad4696_fmc/common/ad4696_bd.tcl

```diff
@@ -0,0 +1,121 @@
+
+create_bd_intf_port -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 ad4696_spi
+
+## To support the 1MSPS (SCLK == 80 MHz), set the spi clock to 160 MHz
+
+ad_ip_instance axi_clkgen spi_clkgen
+ad_ip_parameter spi_clkgen CONFIG.CLK0_DIV 5
+ad_ip_parameter spi_clkgen CONFIG.VCO_DIV 1
+ad_ip_parameter spi_clkgen CONFIG.VCO_MUL 8
+ad_connect $sys_cpu_clk spi_clkgen/clk
+ad_connect spi_clk spi_clkgen/clk_0
+  
+# create a SPI Engine architecture
+
+create_bd_cell -type hier spi_ad4696
+current_bd_instance /spi_ad4696
+
+  create_bd_pin -dir I -type clk clk
+  create_bd_pin -dir I -type rst resetn
+  create_bd_pin -dir I -type clk spi_clk
+  create_bd_pin -dir O irq
+  create_bd_intf_pin -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 m_spi
+  create_bd_intf_pin -mode Master -vlnv xilinx.com:interface:axis_rtl:1.0 M_AXIS_SAMPLE
+
+  if {$adc_resolution <= 32} {
```

Comment:
Do we really need this? The part has a fix 16 bits resolution.

Reply:
this was copied from ad40xx. could probably be improved.